test/e2e/customconfigs/skipjobswithoutqueuename_test.go

```diff
@@ -134,5 +138,122 @@ var _ = ginkgo.Describe("ManageJobsWithoutQueueName", ginkgo.Ordered, func() {
 				}, util.Timeout, util.Interval).Should(gomega.Succeed())
 			})
 		})
+
+		ginkgo.It("should not suspend child jobs of admitted jobs", func() {
+			numPods := 2
+			aw := awtesting.MakeAppWrapper("aw-child", ns.Name).
+				Component(testingjob.MakeJob("job-0", ns.Name).
+					Request(corev1.ResourceCPU, "100m").
+					Parallelism(int32(numPods)).
+					Completions(int32(numPods)).
+					Suspend(false).
+					// Give it enough time that the children would be suspended if parent detection doesn't work as intended.
+					Image(util.E2eTestSleepImage, []string{"3s"}).
+					SetTypeMeta().Obj()).
+				Suspend(false).
+				Obj()
+
+			ginkgo.By("creating an unsuspended appwrapper without a queue name", func() {
+				gomega.Expect(k8sClient.Create(ctx, aw)).To(gomega.Succeed())
+			})
+
+			ginkgo.By("verifying that the appwrapper gets suspended", func() {
+				createdAppWrapper := &awv1beta2.AppWrapper{}
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(aw), createdAppWrapper)).To(gomega.Succeed())
+					g.Expect(createdAppWrapper.Spec.Suspend).To(gomega.BeTrue())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("setting the queue-name label", func() {
+				createdAppWrapper := &awv1beta2.AppWrapper{}
+				awLookupKey := types.NamespacedName{Name: aw.Name, Namespace: ns.Name}
+				gomega.Eventually(func() error {
+					if err := k8sClient.Get(ctx, awLookupKey, createdAppWrapper); err != nil {
+						return err
+					}
+					createdAppWrapper.Labels["kueue.x-k8s.io/queue-name"] = "main"
+					return k8sClient.Update(ctx, createdAppWrapper)
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Wait for appwrapper to be unsuspended", func() {
```

Comment:
nit: is the difference in capitalisation intentional? I don't know enough about how to organize ginkgo tests to comment on the correct style, but noticed that Bys which started with "Wait" were consistently capitilized

Reply:
fixed.